tests/test_hierarchy.cc

```diff
@@ -316,45 +317,73 @@ dealii::TrilinosWrappers::SparseMatrix gimme_identity(unsigned int n_local_rows)
   return identity_matrix;
 }
 
-BOOST_AUTO_TEST_CASE(matrix_transpose_matrix_multiply)
+BOOST_AUTO_TEST_CASE(fast_multiply_transpose)
 {
-  //  unsigned int const n_local_rows = 10;
-  //  unsigned int const n_entries_per_row = 3;
-  //  auto A = gimme_a_matrix(n_local_rows, n_entries_per_row);
-  //  auto B = gimme_a_matrix(n_local_rows, n_entries_per_row);
-  //  dealii::TrilinosWrappers::SparseMatrix C(A.locally_owned_range_indices(),
-  //                                           B.locally_owned_range_indices(),
-  //                                           A.get_mpi_communicator());
-  //  mfmg::matrix_transpose_matrix_multiply(C, B, A);
-  //
-  //  dealii::TrilinosWrappers::SparseMatrix
-  //  BT(B.locally_owned_domain_indices(),
-  //                                            B.locally_owned_range_indices(),
-  //                                            B.get_mpi_communicator());
-  //  // auto I = gimme_identity(n_local_rows);
-  //  // B.Tmmult(BT, I);
-  //  dealii::TrilinosWrappers::SparseMatrix
-  //  C_ref(A.locally_owned_range_indices(),
-  //                                               B.locally_owned_range_indices(),
-  //                                               A.get_mpi_communicator());
-  //  // A.mmult(C_ref, BT);
-  //  int error_code = EpetraExt::MatrixMatrix::Multiply(
-  //      A.trilinos_matrix(), false, B.trilinos_matrix(), true,
-  //      const_cast<Epetra_CrsMatrix &>(BT.trilinos_matrix()));
-  //  BOOST_TEST(error_code == 0);
-  //  C_ref.reinit(BT.trilinos_matrix());
-  //
-  //  dealii::LinearAlgebra::distributed::Vector<double> u(
-  //      A.locally_owned_domain_indices(), A.get_mpi_communicator());
-  //  u = 1.;
-  //
-  //  dealii::LinearAlgebra::distributed::Vector<double> v(
-  //      A.locally_owned_domain_indices(), A.get_mpi_communicator());
-  //
-  //  C.vmult(v, u);
-  //  BOOST_TEST(v.l2_norm() > 1.);
-  //
-  //  u = -1.;
-  //  C_ref.vmult_add(v, u);
-  //  BOOST_TEST(v.l2_norm() == 0.);
+  MPI_Comm comm = MPI_COMM_WORLD;
+  auto comm_size = dealii::Utilities::MPI::n_mpi_processes(comm);
+  // For now fast_ap only work on serial
+  if (comm_size == 1)
```

Comment:
Add a else-clause and check that we get the exception because it is not implemented yet.